sympy/combinatorics/matrix_groups.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import print_function
+from sympy.core import Basic
+from math import sin, cos, pi, floor
+import random
+
+def MatrixMultiply(a, b):
+    """Returns the product of two matrices"""
+
+    _arr = [[] for i in range(len(a))]
+    for i in range(len(a)):
+        for j in range(len(b[0])):
+            c = 0
+            for k in range(len(b)):
+                c = c + (a[i][k] * b[k][j])
+            _arr[i].append(c)
+    return _arr
+
+def isPrime(p):
+    if p is 2:
+        return True
+    elif p%2 is 0:
+        return False
+    for i in range(3, p/2 + 2, 2):
+        if (p % i) is 0:
+            return False
+    return True
+
+def spl_linear_group(dim = 2, min_val = 1, max_val = 10):
+    """
+    Returns a Special Linear Group with integer entries and
+    of specified dimension.
+    """
+
+    if dim < 1:
+        raise ValueError("Size must be greater than or equal to 1")
+    elif dim is 1:
+        return [1]
+
+    _spl_linear_group = SplLinearGroup(dim, min_val, max_val)
+    return _spl_linear_group
+
+def spl_orthogonal_group(dim = 2, x = None):
+    """
+    Returns a Special Orthogonal Group of 2x2 matrices with entries as
+    [  cos(x) sin(x)]
+    [ -sin(x) cos(x)]
+    where x is a real number.
+    """
+
+    if dim < 1:
+        raise ValueError("Size must be greater than or equal to 1")
+    elif dim is 1:
+        return [1]
+    _spl_orthogonal_group = SplOrthogonalGroup(dim, x)
+    return _spl_orthogonal_group
+
+def finite_matrix_group(dim = 2, p = 2):
+    """
+    Returns a dim x dim matrix with entries as integer modulo p, where
+    p is a Prime number.
+    """
+    if dim < 1:
+        raise ValueError("Size must be greater than or equal to 1")
+    if not (isPrime(p)):
+        raise ValueError("p must be a Prime Number")
+
+    _finite_matrix_group = FiniteMatrixGroup(dim, p)
+    return _finite_matrix_group
+
+
+class SplLinearGroup(Basic):
+    def __new__(cls, dim, _min, _max):
+        """
+        Generates two dim x dim matrices, one upper triangular
```

Comment:
A single matrix is not a group so it doesn't make sense to call this class SplLinearGroup. Really, what you are doing here is returning a somewhat random invertible matrix over the integers. This is not really adding anything particularly new to what's already present in the matrices module. Have a look at the documentation for matrices: http://docs.sympy.org/latest/modules/matrices/index.html